curriculum/challenges/english/25-front-end-development/lecture-importance-of-accessibility-and-good-html-structure/672a539b887ec68c593cdc4b.md

```diff
@@ -10,6 +10,102 @@ dashedName: what-are-best-practices-for-tables-and-accessibility
+
+```html
+<table>
+  <caption>Our Pets</caption>
+  <thead>
+    <tr>
+      <!-- Column Headers -->
+      <th>Name</th>
+      <th>Age</th>
+      <th>Type</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>Nora</th> <!-- Row Header -->
+      <td>5</td>
+      <td>Dog</td>
+    </tr>
+    <tr>
+      <th>Gino</th> <!-- Row Header -->
+      <td>2</td>
+      <td>Cat</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+Notice that the above code has a `caption` element immediately after the opening `table` element. Then, inside the Table Head (`th`) element, it has the column headers (name, age, and type). In the second and third rows, inside the table body (`tbody`) element, we find the data for each one of our pets. The names of the pets are the row headers because they are inside table header elements (`th`).
+
+Associating the data cells with their corresponding headers is also very important for screen readers. The scope attribute determines if a header is a row header or a column header.
+Screen readers may guess this correctly from the table's structure, but it's usually recommended to explicitly indicate the scope to ensure clarity.
+
+The `scope` attribute has four possible values. The two you will use most often are "col" for column and "row" for row.
+In the code below, you can see that we added the scope attribute to the column and row headers. The three column headers (Name, Age, and Type) have a scope of "col", column.
+
+The two row headers (Nora and Gino) have a scope of "row".
+
+```html
+<table>
+  <caption>Our Pets</caption>
+  <thead>
+    <tr>
+      <!-- Now they have scope -->
+      <th scope="col">Name</th>
+      <th scope="col">Age</th>
+      <th scope="col">Type</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th scope="row">Nora</th>
+      <td>5</td>
+      <td>Dog</td>
+    </tr>
+    <tr>
+      <th scope="row">Gino</th>
+      <td>2</td>
+      <td>Cat</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+If a column or row header spans across multiple cells, the scope will also be applied to each one of the cells individually.
+
+Some screen readers may not be able to interpret tables with complex structures, so you should also try to flatten the table as much as possible to avoid row and column headers that span across multiple cells.
+
+Your goal should always be to make sure users can access this information, even if their screen readers can't handle complex table structures.
+Now, for cell width, it's recommended to avoid using fixed values. You should use relative values instead, like percentages. Also, try to avoid defining cell height. This will allow users to adjust the text size to fit their needs.
+And finally, you should let the browser determine the table width whenever possible, to reduce the need for horizontal scrolling.
```

Comment:
```suggestion
Your goal should always be to make sure users can access this information, even if their screen readers can't handle complex table structures.

Now, for cell width, it's recommended to avoid using fixed values. You should use relative values instead, like percentages. Also, try to avoid defining cell height. This will allow users to adjust the text size to fit their needs.

And finally, you should let the browser determine the table width whenever possible, to reduce the need for horizontal scrolling.
```